game/scripts/vscripts/components/surrender/surrender.lua

```diff
@@ -0,0 +1,147 @@
+if SurrenderManager == nil then
+  Debug.EnabledModules['surrender:*'] = true
+  DebugPrint ( 'Creating new SurrenderManager object.' )
+  SurrenderManager = class({})
+end
+
+local MINIMUM_KILLS_BEHIND = 50
+local REQUIRED_YES_VOTES = {1, 2, 2, 3, 4}
+local TIME_TO_DISPLAY = 10
+local ONE_MINUTE = 60
+local FIVE_MINUTES = 60 * 5
+local lastTimeSurrenderWasCalled
+local lastTimeSurrenderWasCalledByPlayer = {}
+local votes = {}
+local numberOfVotesCast = 0
+local numberOfVotesExpected = 0
+local teamIdToSurrender
+
+function SurrenderManager:Init ()
+  DebugPrint ('Init SurrenderManager')
+  -- Register chat commands
+  ChatCommand:LinkCommand("-surrender", Dynamic_Wrap(SurrenderManager, "CheckSurrenderConditions"), self)
+  CustomGameEventManager:RegisterListener('surrender_result', Dynamic_Wrap(SurrenderManager, 'PlayerVote'))
+end
+
+function SurrenderManager:CheckSurrenderConditions(keys)
+  local teamId = PlayerResource:GetTeam(keys.playerid)
+  local now = GameRules:GetGameTime()
+  if SurrenderManager:ScoreAllowsSurrender(teamId) and
+      SurrenderManager:TimeAllowsSurrender(keys.playerid, now) then
+    lastTimeSurrenderWasCalled = GameRules:GetGameTime()
+    lastTimeSurrenderWasCalledByPlayer[keys.playerid] = now
+    teamIdToSurrender = teamId
+    local timeout = 10
+    local text = "Would you like to surrender?"
```

Comment:
Send a localisation key like "#surrender_prompt" and do `$.Localize(data.text)` in the JavaScript.